### PR TITLE
feat(test): add `URL` field to annotations for hyperlink display

### LINKED ISSUE #30095: [Feature]: Show URL's within annotations as hyperlink
### 🚀 Feature Request

Add functionality to display URL within annotations as hyperlink instead of just as a string

### Example

```
test.describe(
  "Test URL string hyperlink",
  {
    annotation: [
      {
        type: "issue",
        description:
          "https://github.com/microsoft/playwright/issues/new?assignees=&labels=&projects=&template=feature.yml&title=%5BFeature%5D:+",
      },
      {
        type: "Project",
        description: "Part of this project - https://github.com/microsoft/playwright/issues/new?assignees=&labels=&projects=&template=feature.yml&title=%5BFeature%5D:+",
      },
    ],
  },
```

As can be seen in a report for running the above example test, the "issue" description will be fine - and show as a hyperlink - because it's the only part of the string present

However, the Project description won't show as an hyperlink - just as a normal string

Suggest adding functionality to do the below 

```
test.describe(
  "Test URL string hyperlink",
  {
    annotation: [
      {
        type: "issue",
        description:
          "https://github.com/microsoft/playwright/issues/new?assignees=&labels=&projects=&template=feature.yml&title=%5BFeature%5D:+",
      },
      {
        type: "Project",
        url: "https://github.com/microsoft/playwright/issues/new?assignees=&labels=&projects=&template=feature.yml&title=%5BFeature%5D:+"
        description: "Part of this project",
      },
    ],
  },
```

### Motivation

Easier to click on the hyperlink and open up the URL, similar to how Playwright currently caters for only a URL in a string to be hyperlinked, adding support for this would make it even better..

### PR DESCRIPTION
**Pull request by osohyun0224**
related issue: https://github.com/microsoft/playwright/issues/30095

**Description:**
This pull request introduces the ability to include `URL` directly within annotations in Playwright tests, enabling the URLs to be displayed as hyperlinks in test reports.

**Motivation:**
Current annotation functionality only allows for text descriptions, which limits the ability to navigate directly to related resources or issue trackers. By integrating a URL field, we provide a more interactive and informative testing environment.

**Changes:**
Added an optional url field to the Annotation type to accommodate `URL`
Updated type definitions and documentation to reflect the addition.

**Expected Outcome:**
With the addition of the url field, annotations can now directly link to URLs, which are displayed as hyperlinks in test reports. This update improves the functionality of test descriptions and enhances the accessibility of related online resources.

**References:**
Closes https://github.com/microsoft/playwright/issues/30095 - "[Feature]: Show URL's within annotations as hyperlink"

**Code Check**

<details>
<summary>toggle dropdown</summary>
<div markdown="1">

I have written a test to ensure that the changes were applied correctly, and I have verified its functionality through screenshots. I examined whether the test is running as expected by capturing images of the test execution results.

```
it('should not hang on ctrl-click during provisional load', async ({ context, page, server, isWindows, browserName, isLinux }) => {
  it.info().annotations.push({ type: 'issue', url: 'https://github.com/microsoft/playwright/issues/11595' });
....
```

image
</div>
</details>

### CODE REVIEW

**Comment by osohyun0224** [comment]
@microsoft-github-policy-service agree

File: packages/playwright/types/test.d.ts [hunk 1]
```diff
@@ -8120,6 +8120,11 @@ export interface TestInfo {
      * Optional description.
      */
     description?: string;
+
+    /**
+      * Optional URL to provide additional context or link to an issue tracker.
+    */
```
Comment (injae-kim) [inline packages/playwright/types/test.d.ts:8126]:
```suggestion
    /**
     * Optional URL to provide additional context or link to an issue tracker.
     */
```

**Review by pavelfeldman — APPROVED** [review]

**Comment by pavelfeldman** [comment]
This change looks good. Could you follow up with:
- rendering it in the HTML report and other reports
- rendering it in the Trace
- supporting it in the TeleReceiver/Emitter for shard merger?

**Comment by saurabhdes** [comment]
thanks @osohyun0224 for your change, will this change also fix https://github.com/microsoft/playwright/issues/30140 ? or that's a separate issue

**Comment by osohyun0224** [comment]
@saurabhdes Hello :) I checked your issue, and I think it's an issue related to the newly added feature. I kept the title and description as is and added a new url field. But the functional purpose of your issue is the same ;) I am planning to develop additional features suggested by the maintainer.

**Pull request merged**